ADR-144 Realm resolution from connection string (#150)
decentraland/adr: content/ADR-144.md

---
layout: adr

> adr: 144

Reviewer: ```suggestion
adr: 150
```
And also rename the file

Author: The issue was created to park the ID 144

title: Realm resolution from connection string
date: 2022-12-13
status: Review
type: Standards Track
spdx-license: CC0-1.0
authors:
  - menduz
---

## Abstract

<!--
Abstract is a multi-sentence (short paragraph) technical summary. This should be a very terse and human-readable version of the document section. **Someone should be able to read only the abstract to get the gist of what this document is about in its current state.** Abstracts should be always up to date with the current state of the document.
-->

This document describes the Realm resolution algorithms to consistently resolve servers based on a realm connection string. It resembles DNS, and uses the DAO catalysts, ENS and URLs to resolve the realms as described in [ADR-110](https://adr.decentraland.org/adr/ADR-110).

## Context, Reach & Prioritization

<!--
Discuss and go into detail about the subject in question. Make sure you cover:
- Why is this decision important
- The urgency of the decision
- Datapoints and related background information
- Vocabulary and key terms
-->

> Realms ([ADR-110](https://adr.decentraland.org/adr/ADR-110)) are the configurations that define what and how is executed and rendered in a Decentraland Explorer [ADR-102](https://adr.decentraland.org/adr/ADR-102). If we reason by analogy and consider an Explorer like a Web Browser, and the Scenes like WebSites. Then the Realm is the URL of that site.

Reviewer: ```suggestion
Realms ([ADR-110](https://adr.decentraland.org/adr/ADR-110)) are the configurations that define what and how is executed and rendered in a Decentraland Explorer ([ADR-102](https://adr.decentraland.org/adr/ADR-102)). If we reason by analogy and consider an Explorer like a Web Browser, and the Scenes like WebSites. Then the Realm is the URL of that site.
```

## Specification

<!--
The technical specification should describe the syntax and semantics of any new feature.
-->

> Realm are represented as strings. Those strings need to be resolved to a `/about` endpoint to fetch the configuration.

Reviewer: ```suggestion
Realms are represented as strings. Those strings need to be resolved to an `/about` endpoint to fetch the configuration.
```

The mechanism to resolve that endpoint is:

> 1. If the string is an URL, then it uses the URL

Reviewer: Shouldn't it add `/about` at the end? For e.g. for worlds the URL should be `https://my-worlds-server.io/world/my-name.dcl.eth`, not `https://my-worlds-server.io/world/my-name.dcl.eth/about`

2. If the string equals to the name of a DAO catalyst, then it picks the URL of the Catalyst like `https://peer.decentraland.org/about`
3. If the string is an ENS, then it resolves using ENS `dcl.realm` TXT values
   1. If no values are defined for the ENS-DNS and it is a `.dcl.eth` subdomain, then it falls back to the Worlds resolver
4. If this point is reached, then the realm is invalid.

### ENS resolver

NAME and ENS holders can decide where their worlds are hosted. For that purpose, a special record `dcl.realm` can be added both to ENS and NAME tokens and it should resolve to a base URL that is compliant with the [Realm specification (ADR-110)](https://adr.decentraland.org/adr/ADR-110)

### Worlds resolver

> Worlds is a configuration of services that enable deploying Decentraland scenes using NAME, as described in [ADR-111](/adr/ADR-111). To resolve Worlds, an URL in the shape of `https://worlds-content-server.decentraland.org/world/<NAME>.dcl.eth` is used, the `/about` response is compliant with the [Realm specification (ADR-110)](https://adr.decentraland.org/adr/ADR-110)

Reviewer: ```suggestion
Worlds is a configuration of services that enable deploying Decentraland scenes using a NAME, as described in [ADR-111](/adr/ADR-111). To resolve Worlds, a URL in the shape of `https://worlds-content-server.decentraland.org/world/<NAME>.dcl.eth` is used, the `/about` response is compliant with the [Realm specification (ADR-110)](https://adr.decentraland.org/adr/ADR-110)
```
